tpp-app/src/info/InfoService.js

```diff
@@ -0,0 +1,69 @@
+import AsyncStorage from '@react-native-async-storage/async-storage';
+import { FACT_CYCLE } from '../services/utils/constants.js'
+import { getFullCurrentDateString } from "../services/utils/helpers.js"
+
+/**
+ * Retrieves an array containing the last date the user accessed the info screen and the fact number they saw
+ * @returns returns a promise resolving in an array with format ["2022-1-1", "1"]
+ */
+export const GETFactCycle = async () => new Promise(async (resolve, reject) => {
+    try {
+     if (await AsyncStorage.getItem(FACT_CYCLE.FACT_CYCLE_NUM) == null) {
+        resolve(null);
+     } else {
+         value = await AsyncStorage.getItem(FACT_CYCLE.FACT_CYCLE_NUM).then((value) => {
+             console.log("Retrieved Fact Cycle Date and Number");
+             resolve(value);
+         }).catch(() => {
+             console.log("Failed to retrieve Fact Cycle Date and Number");
+             reject(null);
+         })
+     }
+    } catch (e) {
+        console.log(`GETFactCycle error: ${JSON.stringify(e)}`)
+        reject(null);
+    }
+});
+
+/**
+ * Updates FACT_CYCLE_NUM with the currentDate and updated fact number if the current date and the stored date differ
+ * If GETFactCycle is null, instantiates FACT_CYCLE_NUM with the current date and 1 as the first fact number
+ * @returns a Promise resolving when the setItem or mergeItem operation is complete
+ */
+
+export const POSTFactCycle = async () => new Promise(async (resolve, reject) => {
+    try {
+        if (await GETFactCycle() == null) { // if there is currently no fact/date array stored
+            const value = [];
```

Comment:
something nitpicky lol no need to change this at all it's just my OCD coming out but is there a reason we cannot just do `const value = [getFullCurrentDateString(), "1"]` here

Reply:
me at 7 am for a second thought we couldnt lmao